commit 8058411358c8ae9c4066f672c24e6012348415ec
Author: Brice Goglin <Brice.Goglin@inria.fr>
Date:   Fri Mar 18 20:23:41 2016 +0100

    hwloc_dump_hwdata: don't hardwire /var/run, use @runstatedir@

diff --git a/utils/hwloc/Makefile.am b/utils/hwloc/Makefile.am
index 868ea3a..0486a7a 100644
--- a/utils/hwloc/Makefile.am
+++ b/utils/hwloc/Makefile.am
@@ -1,4 +1,4 @@
-# Copyright © 2009-2015 Inria.  All rights reserved.
+# Copyright © 2009-2016 Inria.  All rights reserved.
 # Copyright © 2009-2012, 2014 Université Bordeaux
 # Copyright © 2009-2014 Cisco Systems, Inc.  All rights reserved.
 #
@@ -47,6 +47,7 @@ endif HWLOC_HAVE_LINUX
 hwloc_dump_hwdata_SOURCES = \
 	hwloc-dump-hwdata.c \
 	hwloc-dump-hwdata-knl.c
+hwloc_dump_hwdata_CPPFLAGS = $(AM_CPPFLAGS) -DRUNSTATEDIR=\"@runstatedir@\"
 
 hwloc_calc_SOURCES = \
         hwloc-calc.h \
diff --git a/utils/hwloc/hwloc-dump-hwdata.c b/utils/hwloc/hwloc-dump-hwdata.c
index b372c1f..844dfef 100644
--- a/utils/hwloc/hwloc-dump-hwdata.c
+++ b/utils/hwloc/hwloc-dump-hwdata.c
@@ -1,6 +1,6 @@
 /*
  * Copyright © 2015 Intel, Inc.  All rights reserved.
- * Copyright © 2015 Inria.  All rights reserved.
+ * Copyright © 2015-2016 Inria.  All rights reserved.
  * See COPYING in top-level directory.
  */
 
@@ -15,7 +15,7 @@
 #include <errno.h>
 #include <assert.h>
 
-#define DEFAULT_DUMP_DIR "/var/run/hwloc"
+#define DEFAULT_DUMP_DIR RUNSTATEDIR "/hwloc"
 
 extern int hwloc_dump_hwdata_knl_smbios(const char *input_fsroot, const char *filename);
 
