diff -Nrup hwloc-1.11.2.orig/contrib/systemd/Makefile.am hwloc-1.11.2/contrib/systemd/Makefile.am
--- hwloc-1.11.2.orig/contrib/systemd/Makefile.am	2016-06-06 14:40:16.698823658 -0400
+++ hwloc-1.11.2/contrib/systemd/Makefile.am	2016-06-06 17:19:50.456582250 -0400
@@ -7,7 +7,10 @@
 EXTRA_DIST = hwloc-dump-hwdata.service.in
 
 if HWLOC_HAVE_LINUX
+if HWLOC_HAVE_X86
 nodist_pkgdata_DATA = hwloc-dump-hwdata.service
+endif HWLOC_HAVE_X86
+
 # depend on Makefile in case we rerun configure with different directory arguments
 hwloc-dump-hwdata.service: hwloc-dump-hwdata.service.in Makefile
 	@echo Creating $@...
